signingscript/src/signingscript/sign.py

```diff
@@ -1548,3 +1549,50 @@ async def sign_debian_pkg(context, path, fmt, *args, **kwargs):
             output_file.write(signed_file["content"])
     await _create_tarfile(context, path, all_file_names, compression, tmp_dir=tmp_dir)
     return path
+
+
+async def rcodesign_notarize(paths, notarization_workdir, creds_path):
+    NOTARIZATION_EXTENSIONS = ("app", "pkg")
+    for app in paths:
+        if app.split(".")[-1].lower() not in NOTARIZATION_EXTENSIONS:
+            raise SigningScriptError("Unable to notarize app format: %s" % app)
+        app_path = os.path.join(notarization_workdir, app)
+        command = [
+            "rcodesign",
+            "notary-submit",
+            "--staple",
+            "--api-key-path",
+            creds_path,
+            app_path,
+        ]
+        await utils.execute_subprocess(command)
+
+
+@time_async_function
+async def apple_notarize(context, path, *args, **kwargs):
+    """
+    Notarizes given package(s) using rcodesign.
+    """
+    # Get credentials
+    cert_type = task.task_cert_type(context)
+    if cert_type not in context.apple_notarization_configs:
+        raise SigningScriptError("Credentials not found for scope: %s" % cert_type)
+    scope_credentials =  context.apple_notarization_configs.get(cert_type)
+    if len(scope_credentials) != 1:
+        raise SigningScriptError("There should only be 1 scope credential, %s found." % len(scope_credentials))
+    credential = asdict(scope_credentials[0])
+    
+    # Setup workdir
+    notarization_workdir = os.path.join(context.config["work_dir"], "apple_notarize")
+    _, compression = os.path.splitext(path)
+    all_file_names = await _extract_tarfile(context, path, compression, tmp_dir=notarization_workdir)
+    workdir_files = os.listdir(notarization_workdir)
+
+    # Notarize
+    with tempfile.NamedTemporaryFile() as temp_creds:
+        temp_creds.write(json.dumps(credential).encode("ascii"))
```

Comment:
Is there a reason we can't simply deploy creds as this, rather than write them out with each task? We do this, for example, with Google credentials in pushapk workers: [creating files as part of init](https://github.com/mozilla-releng/scriptworker-scripts/blob/6f19b279ee606ebdb5c155f3a1afacb44be6d5e4/pushapkscript/docker.d/init_worker.sh#L69-L79), [setting the file path as part of the worker config](https://github.com/mozilla-releng/scriptworker-scripts/blob/master/pushapkscript/docker.d/worker.yml#L190)

Reply:
I'm not sure I love the idea of either:
a. Having the credentials in SOPS base64'd
b. Having to construct the json file in `init_worker.sh`

What if we moved the credentials creation to `async_main`? (`if !credential.exist, create it` sort of deal)

Reply:
It will have to be in SOPS either way, no? (I don't think we have any other place to put secrets that these workers need?)

As far as init_worker vs. async_main - I don't feel terribly strongly either way. Although requiring an intermediary format to `async_main` makes local testing a bit more annoying? I'm not going to block on either approach - as long as it's not being done per-task I'm pretty happy.